File: frontend/src/types/vehicleOwnerContract.ts
export type ContractStatus = 'draft' | 'sent';

export interface VehicleOwnerContract {
  id: number;
  owner_id: number;
  contract_number: string;
  status: ContractStatus;
  pdf_path: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/vehicle_owner_contract.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel

from app.models.vehicle_owner_contract import ContractStatus


class VehicleOwnerContractRead(BaseModel):
    id: int
    owner_id: int
    contract_number: str
    status: ContractStatus
    pdf_path: Optional[str] = None
    notes: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
